### PR TITLE
adding shared base types for retrievable resource action logs

### PR DESCRIPTION
**Pull request by nmorasb**
## MADiE PR Template

Jira Ticket: [MAT-4470](https://jira.cms.gov/browse/MAT-4470)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [ ] Required tests are included.
* [x] No extemporaneous files are included (i.e. Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/models/common/ActionType.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package gov.cms.madie.models.common;
+
+public enum ActionType {
+  CREATED,
+  VERSIONED_MAJOR,
+  VERSIONED_MINOR,
+  DELETED
```
[the remark this answers is not in the document]
Comment (nmorasb) [reply src/main/java/gov/cms/madie/models/common/ActionType.java:7]: As of right now, only these were requested, I actually split versioned to give a little more info. Nothing was indicated about tracking general updates.

1. Store when (date/time) and by whom (HARP ID) when a life cycle action is taken against a library. Life cycle actions are defined as when a library is:
    *  versioned
    *  created

**Review by adongare — APPROVED** [review]
